packages/vite/src/node/server/pluginContainer.ts

```diff
@@ -578,7 +578,16 @@ export async function createPluginContainer(
           break
         }
         if (!combinedMap) {
-          combinedMap = m as SourceMap
+          const sm = m as SourceMap
+          if (sm.sources.length === 1 && !sm.sources[0]) {
```

Comment:
```suggestion
          // sourcemap should not include `sources: [null]` (because `sources` should be string) nor
          // `sources: ['']` (because `''` means the path of sourcemap)
          // but MagicString generates this when `filename` option is not set.
          // Rollup supports these and therefore we support this as well
          if (sm.sources.length === 1 && !sm.sources[0]) {
```
Let's make it clear why we need to support this.